core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -876,8 +888,15 @@ class DAGScheduler(
       // SparkListenerStageCompleted here in case there are no tasks to run.
       outputCommitCoordinator.stageEnd(stage.id)
       listenerBus.post(SparkListenerStageCompleted(stage.latestInfo))
-      logDebug("Stage " + stage + " is actually done; %b %d %d".format(
-        stage.isAvailable, stage.numAvailableOutputs, stage.numPartitions))
+
+      val debugString = stage match {
+        case stage: ShuffleMapStage =>
+          "Stage " + stage + " is actually done; %b %d %d".format(stage.isAvailable,
```

Comment:
As long as you're editing this line, would you mind making this log message slightly more readable"

s"Stage $stage is actually done (available: ${stage.isAvailable}, available outputs: ${stage.num..}, partitions: ${stage.numPartitions)"